File: src/main/java/no/nav/pensjon/regler/domain/beregning2011/FremskrivingsDetaljer.java
package no.nav.pensjon.regler.domain.beregning2011;

import java.io.Serializable;
import java.util.Date;

public class FremskrivingsDetaljer implements Serializable, IFremskriving {

    private static final long serialVersionUID = 1574525081526453384L;

    private Date justeringTomDato;
    private double justeringsfaktor;
    private double teller;
    private double nevner;
    private int arskull;

    public FremskrivingsDetaljer() {
        super();
    }

    public FremskrivingsDetaljer(FremskrivingsDetaljer frem) {
        this();
        if (frem.justeringTomDato != null) {
            justeringTomDato = (Date) frem.justeringTomDato.clone();
        }
        justeringsfaktor = frem.justeringsfaktor;
        teller = frem.teller;
        nevner = frem.nevner;
        arskull = frem.arskull;
    }

    @Override
    public double getJusteringsfaktor() {
        return justeringsfaktor;
    }

    @Override
    public void setJusteringsfaktor(double justeringsfaktor) {
        this.justeringsfaktor = justeringsfaktor;
    }

    @Override
    public Date getJusteringTomDato() {
        return justeringTomDato;
    }

    @Override
    public void setJusteringTomDato(Date justeringTomDato) {
        this.justeringTomDato = justeringTomDato;
    }

    @Override
    public double getNevner() {
        return nevner;
    }

    @Override
    public void setNevner(double nevner) {
        this.nevner = nevner;
    }

    @Override
    public double getTeller() {
        return teller;
    }

    @Override
    public void setTeller(double teller) {
        this.teller = teller;
    }

    public int getArskull() {
        return arskull;
    }

    public void setArskull(int arskull) {
        this.arskull = arskull;
    }
}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning2011/FremskrivingsDetaljer.kt
package no.nav.pensjon.regler.domain.beregning2011

import java.io.Serializable
import java.util.*
import java.time.LocalDate

class FremskrivingsDetaljer : Serializable, IFremskriving {
    @Deprecated("Use justeringTomDatoLd instead")
    override var justeringTomDato: Date? = null
    override var justeringTomDatoLd: LocalDate? = null
    override var justeringsfaktor = 0.0
    override var teller = 0.0
    override var nevner = 0.0
    var arskull = 0

    constructor()
    constructor(frem: FremskrivingsDetaljer) : this() {
        if (frem.justeringTomDato != null) {
            justeringTomDato = frem.justeringTomDato!!.clone() as Date
        }
        if(frem.justeringTomDatoLd != null) {
            justeringTomDatoLd = frem.justeringTomDatoLd
        }
        justeringsfaktor = frem.justeringsfaktor
        teller = frem.teller
        nevner = frem.nevner
        arskull = frem.arskull
    }
}